docs/r2/image_summaries.ipynb

```diff
@@ -0,0 +1,639 @@
+        ")"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "SdT_PpZB1UMn",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "When training a classifier, it's useful to see the [confusion matrix](https://en.wikipedia.org/wiki/Confusion_matrix). The confusion matrix gives you detailed knowledge of how your classifier is performing on test data.\n",
+        "\n",
+        "Define a function that calculates the confusion matrix. You'll use a convenient [Scikit-learn](https://https://scikit-learn.org/stable/auto_examples/model_selection/plot_confusion_matrix.html) function to do this, and then plot it using matplotlib."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "rBiXP8-UO8t6",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "def plot_confusion_matrix(cm, class_names):\n",
+        "  \"\"\"\n",
+        "  Returns a matplotlib figure containing the plotted confusion matrix.\n",
+        "\n",
+        "  Args:\n",
+        "    cm (array, shape = [n, n]): a confusion matrix of integer classes\n",
+        "    class_names (array, shape = [n]): String names of the integer classes\n",
+        "  \"\"\"\n",
+        "  figure = plt.figure(figsize=(8, 8))\n",
+        "  plt.imshow(cm, interpolation='nearest', cmap=plt.cm.Blues)\n",
+        "  plt.title(\"Confusion matrix\")\n",
+        "  plt.colorbar()\n",
+        "  tick_marks = np.arange(len(class_names))\n",
+        "  plt.xticks(tick_marks, class_names, rotation=45)\n",
+        "  plt.yticks(tick_marks, class_names)\n",
+        "\n",
+        "  # Normalize the confusion matrix.\n",
+        "  cm = np.around(cm.astype('float') / cm.sum(axis=1)[:, np.newaxis], decimals=2)\n",
+        "\n",
+        "  # Use white text if squares are dark; otherwise black.\n",
+        "  threshold = cm.max() / 2.\n",
+        "  for i, j in itertools.product(range(cm.shape[0]), range(cm.shape[1])):\n",
+        "    color = \"white\" if cm[i, j] > threshold else \"black\"\n",
+        "    plt.text(j, i, cm[i, j], horizontalalignment=\"center\", color=color)\n",
+        "\n",
+        "  plt.tight_layout()\n",
+        "  plt.ylabel('True label')\n",
+        "  plt.xlabel('Predicted label')\n",
+        "  return figure"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "6lOAl_v26QGq",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "You're now ready to train the classifier and regularly log the confusion matrix along the way.\n",
+        "\n",
+        "Here's what you'll do:\n",
+        "\n",
+        "1. Create the Keras TensorBoard callback to log basic metrics\n",
+        "2. Create a Keras LambdaCallback to log the confusion matrix at the end of every epoch\n",
```

Comment:
Might be worth linking to the docs for this as it hasn't been encountered yet in the TensorBoard docs.

Reply:
Done.